nsv13/code/modules/munitions/ship_weapons/energy_weapons/bsa.dm

```diff
@@ -186,8 +186,27 @@
 	new /obj/effect/temp_visual/bsa_splash(point, dir)
 	//Recharging...
 	get_overmap()?.relay_to_nearby(weapon_type.overmap_select_sound)
-
 	if(blocker)
+		var/turf/location = get_turf(blocker)
+		var/admin_message = "The ship BSA has hit [blocker] at [ADMIN_VERBOSEJMP(location)]!"
+		var/log_message = "The ship BSA has hit [blocker] at [AREACOORD(location)]!"
+		var/firer = get_overmap().pilot
+		if(!firer)
+			firer = src
+			admin_message += " BSA fired by [firer]!"
+		else
+			admin_message += " BSA fired by [ADMIN_LOOKUPFLW(firer)]!"
+		if(istype(blocker, /obj/machinery/the_singularitygen)) //What did you do
+			var/evilperson = get_mob_by_ckey(blocker.fingerprintslast)
+			GLOB.bombers += evilperson
+			admin_message += " [blocker] was last touched by [ADMIN_LOOKUPFLW(evilperson)]!"
+			log_message += " [blocker] was last touched by [key_name_admin(evilperson)]!"
+			new /obj/anomaly/singularity(location, 2500)
+			var/turf/overmaplocation = get_turf(get_overmap())
+			if(overmaplocation)
+				new /obj/effect/overmap_anomaly/singularity(overmaplocation)
```

Comment:
Should link this anomaly to the ship's current system (and probably just drop it into the aether if in hyperspace?) or it'll not do anything (due to the change making them use system contents)

Reply:
I figured it wouldn't be necessary because this will instantly end the round

Reply:
Weeeelll wouldn't since overmap singularities go by system contents to see what they affect in what way.. which, if there isn't any present, the ship is going to be perfectly fine (but I guess you should should call the overmap sungularities procs instead if you do not feel like linking it)